File: normal-soves/MiddleoftheLinkedList.java
class ListNode {
    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

public class MiddleoftheLinkedList {
    public ListNode middleNode(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }

        ListNode t = head, r = head;
        while (r != null && r.next != null) {
            t = t.next;
            r = r.next.next;
        }
        return t;
    }
}

File: A2SV-G7/Problems/middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count = 0
        curr = head
        while curr:
            count += 1
            curr = curr.next
        mid = (count // 2) + 1
        print(mid)
        mid_node = None
        curr = head
        while curr:
            print(curr.val)
            if mid == 1:
                return curr
            curr = curr.next
            mid -= 1
